p2panda-net/src/sync/mod.rs

```diff
@@ -105,18 +113,14 @@ where
     }
 
     // Run the sync protocol.
-    let result = sync_protocol
+    sync_protocol
         .initiate(
             topic.clone(),
             Box::new(&mut send),
             Box::new(&mut recv),
             Box::new(&mut sink),
         )
-        .await;
-
-    if let Err(err) = result {
-        error!("sync protocol initiation failed: {err}");
-    }
+        .await?;
```

Comment:
There's a slight asymmetry here, in that `initiate_sync()` returns the error to the caller (`connect_and_sync()` in the manager) whereas `accept_sync()` uses the oneshot channel to inform the engine actor more directly.

I'm not unhappy about the slight difference, since returning the error in `initiate` allows us to match on connection vs. sync errors and take appropriate action (rescheduling an attempt or informing the engine actor).

Reply:
An comment in this place might be nice!